src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -35,47 +35,213 @@
 const uint32_t kThreadPoolSize = 1;
 const bool kHmiLaunch = false;
 const uint32_t kStopStreamingTimeout = 10000;
+const uint32_t kWaitFreeQueue = 2000;
+
+bool is_called_ = false;
+uint32_t hmi_app_id_ = 0u;
+sync_primitives::Lock state_lock_;
+sync_primitives::ConditionalVariable state_condition_;
 }  // namespace
 
 class ApplicationManagerImplTest : public ::testing::Test {
  protected:
   virtual void SetUp() OVERRIDE {
-    app_manager_impl.reset(new am::ApplicationManagerImpl(
-        mock_application_manager_settings_, mock_policy_settings_));
+    const uint32_t timeout = 10000u;
+    std::string directory_name = "./test_storage";
+    const std::string folder_name = "app";
+    const std::string test_path =
+        file_system::ConcatPath(directory_name, folder_name);
+    const std::string file1 =
+        file_system::ConcatPath(test_path, "test_file2.bmp");
+
+    ASSERT_TRUE(file_system::CreateDirectoryRecursively(test_path));
+    ASSERT_TRUE(file_system::CreateFile(file1));
+
+    ON_CALL(mock_application_manager_settings, thread_pool_size())
+        .WillByDefault(Return(2u));
+    ON_CALL(mock_application_manager_settings, app_icons_folder())
+        .WillByDefault(ReturnRef(directory_name));
+    EXPECT_CALL(mock_application_manager_settings, app_storage_folder())
+        .WillRepeatedly(ReturnRef(directory_name));
+    ON_CALL(mock_application_manager_settings, launch_hmi())
+        .WillByDefault(Return(true));
+    ON_CALL(mock_application_manager_settings, stop_streaming_timeout())
+        .WillByDefault(Return(1u));
+    ON_CALL(mock_application_manager_settings, default_timeout())
+        .WillByDefault(ReturnRef(timeout));
+    app_manager_impl_.reset(new am::ApplicationManagerImpl(
+        mock_application_manager_settings, mock_policy_settings));
 
     app_id_ = 10;
+    const std::string kMacAddr_("kMacAddr_ess");
+
+    ON_CALL(mock_session_observer, GetDataOnDeviceID(_, _, _, _, _))
+        .WillByDefault(DoAll(SetArgPointee<3u>(kMacAddr_), Return(0)));
+
+    ON_CALL(mock_connection_handler, GetDataOnSessionKey(_, _, _, _))
+        .WillByDefault(DoAll(SetArgPointee<3u>(app_id_), Return(0)));
+    ON_CALL(mock_connection_handler, get_session_observer())
+        .WillByDefault(ReturnRef(mock_session_observer));
+
+    mock_storage =
+        ::utils::MakeShared<NiceMock<resumption_test::MockResumptionData>>(
+            app_mngr_);
+
+    get_app_manager().resume_controller().set_resumption_storage(mock_storage);
+    get_app_manager().set_connection_handler(&mock_connection_handler);
+    get_app_manager().set_hmi_message_handler(&mock_hmi_mess_handler);
   }
 
   uint32_t app_id_;
   MockApplicationManagerSettings mock_application_manager_settings_;
   policy_test::MockPolicySettings mock_policy_settings_;
 
   am::ApplicationManagerImpl& get_app_manager() {
-    return (*app_manager_impl.get());
+    return (*app_manager_impl_.get());
```

Comment:
what the purpose of this method?
Looks like we return refference to the pointer which we store in the same class.